internal/keepers/ocr.go

```diff
@@ -55,7 +63,7 @@ func (k *keepers) Observation(ctx context.Context, rt types.ReportTimestamp, _ t
 
 	// keyList produces a sorted result so the following reduction of keys
 	// should be more uniform for all nodes
-	keys := keyList(filterUpkeeps(results, ktypes.Eligible))
+	keys := keyList(filterUpkeeps(results, ktypes.Eligible, observationKeysLimit))
```

Comment:
not sure if it makes sense to expand scope of filter. why not a separate step after filter?

Reply:
Makes sense. Done.